src/lammpsio/snapshot.py

```diff
@@ -176,31 +222,122 @@ def to_hoomd_gsd(self, type_map=None):
                 for typeidx, typeid in enumerate(sorted_typeids):
                     frame.particles.typeid[self.typeid == typeid] = typeidx
             else:
-                frame.particles.types = list(type_map.values())
-                reverse_type_map = {
-                    typeid: typeidx for typeidx, typeid in enumerate(type_map.keys())
-                }
-                for i, typeid in enumerate(self.typeid):
-                    frame.particles.typeid[i] = reverse_type_map[typeid]
+                if type_map is not None:
+                    warnings.warn(
+                        "type_map will be deprecated, use label instead.",
+                        DeprecationWarning,
+                        stacklevel=2,
+                    )
+                    type_label_map = LabelMap(map=type_map)
+                else:
+                    type_label_map = self.label
+                frame.particles.types = type_label_map.types
+                Snapshot._set_type_id(
+                    self,
+                    self.typeid,
+                    frame.particles.typeid,
+                    self.label,
```

Comment:
Does this need to be this?
```suggestion
                    type_label_map,
```

Make sure there is a unit test that covers this case.